src/argus_htmx/destinations/forms.py

```diff
@@ -0,0 +1,79 @@
+from django import forms
+from django.forms import ModelForm
+
+from argus.notificationprofile.models import DestinationConfig, Media
+from argus.notificationprofile.serializers import RequestDestinationConfigSerializer
+from argus.notificationprofile.media import api_safely_get_medium_object
+
+
+class DestinationFormCreate(ModelForm):
+    settings = forms.CharField(required=True)
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request", None)
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = DestinationConfig
+        fields = ["label", "media", "settings"]
+        labels = {
+            "label": "Name",
+        }
+
+    def clean(self):
+        super().clean()
+        settings_key = _get_settings_key_for_media(self.cleaned_data["media"])
+        self.cleaned_data["settings"] = {settings_key: self.cleaned_data["settings"]}
```

Comment:
Add comment to explain that this is for making the value compatible with JSONField, which is what DestinationConfig.settings should really be. The form overrides it to be a CharField for the frotnend to make more sense (we shouldnt expect the user to give a full valid dict in JSON format as input, only the email address / phone number itself should be requires)